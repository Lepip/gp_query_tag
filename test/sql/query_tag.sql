CREATE EXTENSION query_tag;


SET QUERY_TAG TO "aboba=aboba";

-- weird query_tags

SET QUERY_TAG TO "a";
SHOW QUERY_TAG;

SET QUERY_TAG TO "=aboba";
SHOW QUERY_TAG;

SET QUERY_TAG TO "a=a;";
SHOW QUERY_TAG;

SET QUERY_TAG TO "a;";
SHOW QUERY_TAG;

SET QUERY_TAG TO "123=123";
SHOW QUERY_TAG;

SET QUERY_TAG TO ";";
SHOW QUERY_TAG;

-- complex behaviour

-- should set to rgroup1

CREATE RESOURCE GROUP rgroup1 WITH (CPU_RATE_LIMIT=20, MEMORY_LIMIT=25, MEMORY_SPILL_RATIO=20);
CREATE RESOURCE GROUP rgroup2 WITH (CPU_RATE_LIMIT=20, MEMORY_LIMIT=25, MEMORY_SPILL_RATIO=20);

INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('admin_group', 'clepip', 'rgroup1', 3, 'group=rgroup1', TRUE, FALSE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('admin_group', 'clepip', 'rgroup2', 4, 'group=rgroup1', TRUE, FALSE);

SET QUERY_TAG TO "group=rgroup1";
SELECT current_resgroup();

-- should still set to rgroup1 (we are in admin_group, the "current_rsgroup" shows group after
--- applying tags, and tags are applied accordingly to users rsgroup)

INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('rgroup1', 'clepip', 'rgroup2', 1, 'group=still_1', TRUE, FALSE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('admin_group', 'clepip', 'rgroup1', 5, 'group=still_1', TRUE, FALSE);

SET QUERY_TAG TO "group=still_1";
SELECT CURRENT_RESGROUP();

-- should set to rgroup1

INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('admin_group', 'clepip', 'rgroup2', 1, 'group=cancelled', FALSE, FALSE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule)
    VALUES ('admin_group', 'clepip', 'rgroup1', 2, 'group=cancelled', TRUE, FALSE);

SET QUERY_TAG TO "group=cancelled";
SELECT CURRENT_RESGROUP();

INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active, kill_rule, planner_cost, orca_cost)
    VALUES ('admin_group', 'clepip', 'rgroup1', 2, 'group=cancelled', FALSE, TRUE, 1000, 1000);