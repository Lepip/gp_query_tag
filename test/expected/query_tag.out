CREATE EXTENSION query_tag;
SET QUERY_TAG TO "aboba=aboba";
-- weird query_tags
SET QUERY_TAG TO "a";
ERROR:  invalid value for parameter "QUERY_TAG": "a"
SHOW QUERY_TAG;
  QUERY_TAG  
-------------
 aboba=aboba
(1 row)

SET QUERY_TAG TO "=aboba";
ERROR:  invalid value for parameter "QUERY_TAG": "=aboba"
SHOW QUERY_TAG;
  QUERY_TAG  
-------------
 aboba=aboba
(1 row)

SET QUERY_TAG TO "a=a;";
ERROR:  invalid value for parameter "QUERY_TAG": "a=a;"
SHOW QUERY_TAG;
  QUERY_TAG  
-------------
 aboba=aboba
(1 row)

SET QUERY_TAG TO "a;";
ERROR:  invalid value for parameter "QUERY_TAG": "a;"
SHOW QUERY_TAG;
  QUERY_TAG  
-------------
 aboba=aboba
(1 row)

SET QUERY_TAG TO "123=123";
SHOW QUERY_TAG;
 QUERY_TAG 
-----------
 123=123
(1 row)

SET QUERY_TAG TO ";";
ERROR:  invalid value for parameter "QUERY_TAG": ";"
SHOW QUERY_TAG;
 QUERY_TAG 
-----------
 123=123
(1 row)

-- complex behaviour
-- should set to rgroup1
CREATE RESOURCE GROUP rgroup1 WITH (CPU_RATE_LIMIT=20, MEMORY_LIMIT=25, MEMORY_SPILL_RATIO=20);
CREATE RESOURCE GROUP rgroup2 WITH (CPU_RATE_LIMIT=20, MEMORY_LIMIT=25, MEMORY_SPILL_RATIO=20);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('admin_group', 'clepip', 'rgroup1', 3, 'group=rgroup1', TRUE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('admin_group', 'clepip', 'rgroup2', 4, 'group=rgroup1', TRUE);
SET QUERY_TAG TO "group=rgroup1";
SELECT current_resgroup();
 current_resgroup 
------------------
 rgroup1
(1 row)

-- should still set to rgroup1 (we are in admin_group, the "current_rsgroup" shows group after
--- applying tags, and tags are applied accordingly to users rsgroup)
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('rgroup1', 'clepip', 'rgroup2', 1, 'group=still_1', TRUE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('admin_group', 'clepip', 'rgroup1', 5, 'group=still_1', TRUE);
SET QUERY_TAG TO "group=still_1";
SELECT CURRENT_RESGROUP();
 current_resgroup 
------------------
 rgroup1
(1 row)

-- should set to rgroup1
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('admin_group', 'clepip', 'rgroup2', 1, 'group=cancelled', FALSE);
INSERT INTO wlm_rules (resgname, role, dest_resg, order_id, query_tag, active)
    VALUES ('admin_group', 'clepip', 'rgroup1', 2, 'group=cancelled', TRUE);
SET QUERY_TAG TO "group=cancelled";
SELECT CURRENT_RESGROUP();
 current_resgroup 
------------------
 rgroup1
(1 row)

